appengine/standard/memcache/guestbook/main.py

```diff
@@ -85,8 +85,11 @@ def get_greetings(self, guestbook_name):
         greetings = memcache.get('{}:greetings'.format(guestbook_name))
         if greetings is None:
             greetings = self.render_greetings(guestbook_name)
-            if not memcache.add('{}:greetings'.format(guestbook_name),
-                                greetings, 10):
+            try:
+                if not memcache.add('{}:greetings'.format(guestbook_name),
+                                    greetings, 10):
+                    logging.error('Memcache set failed.')
```

Comment:
Added a more specific message for data larger than 1MB.

For the memcache.add False result, it might not necessarily be a key already exists error, so I'm leaving it as 'Memcache set failed.'